### PR TITLE
Set entity_guid after creating NRQL alert condition

### LINKED ISSUE #2150: newrelic_nrql_alert_contidion entity_guid not set on creation
# Computed fields on object creation not being set:

https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_newrelic_nrql_alert_condition.go#L436 you set the id but not the entity guid at creation so it cannot be used in the same apply it was created. This causes partial terraform applies. As the read will correctly pull the entity guid, but not the initial create.  

### Terraform Version
Any

### Affected Resource(s)
Please list the resources as a list, for example:
- `newrelic_nrql_alert_condition`

### Terraform Configuration
```hcl
resource "newrelic_alert_policy" "foo" {
  name = "foo"
}


resource "newrelic_nrql_alert_condition" "foo" {
  account_id                     = <Your Account ID>
  policy_id                      = newrelic_alert_policy.foo.id
  type                           = "static"
  name                           = "foo"
  description                    = "Alert when transactions are taking too long"
  runbook_url                    = "https://www.example.com"
  enabled                        = true
  violation_time_limit_seconds   = 3600
  fill_option                    = "static"
  fill_value                     = 1.0
  aggregation_window             = 60
  aggregation_method             = "event_flow"
  aggregation_delay              = 120
  expiration_duration            = 120
  open_violation_on_expiration   = true
  close_violations_on_expiration = true
  slide_by                       = 30

  nrql {
    query = "SELECT average(duration) FROM Transaction where appName = 'Your App'"
  }

  critical {
    operator              = "above"
    threshold             = 5.5
    threshold_duration    = 300
    threshold_occurrences = "ALL"
  }

  warning {
    operator              = "above"
    threshold             = 3.5
    threshold_duration    = 600
    threshold_occurrences = "ALL"
  }
}

resource "newrelic_entity_tags" "foo" {
  guid = newrelic_nrql_alert_condition.foo.entity_guid

  tag {
    key    = "my-key"
    values = ["my-value", "my-other-value"]
  }

  tag {
    key    = "my-key-2"
    values = ["my-value-2"]
  }
}
```

### Actual Behavior
What actually happened?
entity_guid null

### Expected Behavior
What should have happened?
entity_guid is actual guid 

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### PR DESCRIPTION
**Pull request by aw129**
Fixes issue created in https://github.com/newrelic/terraform-provider-newrelic/pull/2082 where the `entity_guid` property is no longer set when a newrelic_nrql_alert_condition has been created.

# Description

Restores the call to `flattenNrqlAlertCondition` which was lost when `resourceNewRelicAlertPolicyChannelRead` stopped being called. `flattenNrqlAlertCondition` sets a number of properties on the resource including the `entity_guid`.

Fixes #2150 #2115

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

Please delete options that are not relevant.

- [X] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [X] My code is formatted to [Go standards](https://go.dev/blog/gofmt)
- [X] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes. Go [here](https://github.com/newrelic/terraform-provider-newrelic#testing) for instructions on running tests locally.

## How to test this change?

Please describe how to test your changes. Include any relevant steps in the UI, HCL file(s), commands, etc

- Create a `newrelic_nrql_alert_condition` and assign tags to it using `newrelic_entity_tags`
  ```
  resource "newrelic_alert_policy" "foo" {
    name = "foo"
  }
  
  
  resource "newrelic_nrql_alert_condition" "foo" {
    account_id                     = <Your Account ID>
    policy_id                      = newrelic_alert_policy.foo.id
    type                           = "static"
    name                           = "foo"
    description                    = "Alert when transactions are taking too long"
    runbook_url                    = "https://www.example.com"
    enabled                        = true
    violation_time_limit_seconds   = 3600
    fill_option                    = "static"
    fill_value                     = 1.0
    aggregation_window             = 60
    aggregation_method             = "event_flow"
    aggregation_delay              = 120
    expiration_duration            = 120
    open_violation_on_expiration   = true
    close_violations_on_expiration = true
    slide_by                       = 30
  
    nrql {
      query = "SELECT average(duration) FROM Transaction where appName = 'Your App'"
    }
  
    critical {
      operator              = "above"
      threshold             = 5.5
      threshold_duration    = 300
      threshold_occurrences = "ALL"
    }
  
    warning {
      operator              = "above"
      threshold             = 3.5
      threshold_duration    = 600
      threshold_occurrences = "ALL"
    }
  }
  
  resource "newrelic_entity_tags" "foo" {
    guid = newrelic_nrql_alert_condition.foo.entity_guid
  
    tag {
      key    = "my-key"
      values = ["my-value", "my-other-value"]
    }
  
    tag {
      key    = "my-key-2"
      values = ["my-value-2"]
    }
  }
  ```

### CODE REVIEW

File: newrelic/resource_newrelic_nrql_alert_condition.go [hunk 1]
```diff
@@ -435,7 +435,7 @@ func resourceNewRelicNrqlAlertConditionCreate(ctx context.Context, d *schema.Res
```
Comment (mbazhlekova) [inline newrelic/resource_newrelic_nrql_alert_condition.go:435]:
```suggestion
d.Set("entity_guid", condition.EntityGUID)
```

**Comment by mbazhlekova** [comment]
Hi @aw129 👋 thank you for your contribution! We made the decision not to return the read from the create function because it caused other problems. I see that we've introduced this issue with `entity_guid` however. Perhaps we can just set the `entity_guid` on create instead of returning the read like my comment suggests.

**Comment by aw129** [comment]
Hi @mbazhlekova,

I didn't call the read function (`resourceNewRelicNrqlAlertConditionRead`) but the `flattenNrqlAlertCondition` one instead (which the read ultimately calls). This sets many properties on the ResourceData and I thought calling this would help to avoid introducing similar inconsistencies in the future. Only one place needs to be updated if any new properties are added.

Happy to just set the `entity_guid` directly in the create method if you think that's best.

**Comment by mbazhlekova** [comment]
> Hi @mbazhlekova,
> 
> I didn't call the read function (`resourceNewRelicNrqlAlertConditionRead`) but the `flattenNrqlAlertCondition` one instead (which the read ultimately calls). This sets many properties on the ResourceData and I thought calling this would help to avoid introducing similar inconsistencies in the future. Only one place needs to be updated if any new properties are added.
> 
> Happy to just set the `entity_guid` directly in the create method if you think that's best.

Ah I see. My brain might have just filled in the read function there. I think it should be ok if we return the flatten.

**Review by mbazhlekova — APPROVED** [review]

**Comment by aw129** [comment]
@mbazhlekova is there something more I need to do about the "license/cla" check? I have signed the CLA.

**Pull request merged**